examples/00-basic/13-get_data_by_id.py

```diff
@@ -0,0 +1,173 @@
+mesh_from_scoping_op.inputs.mesh.connect(model.metadata.meshed_region)
+mesh_selection = mesh_from_scoping_op.outputs.mesh()
+
+###############################################################################
+# Plot ``mesh_selection``
+mesh_selection.plot()
+
+###############################################################################
+# The plot shows only one of the two cubes from the total geometry shown before.
+
+###############################################################################
+# Identify the location of the Field
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Check the location of the scoping to see what is the information associated
+# to the IDs. The displacement field is defined at ``"nodal"`` location. Other
+# fields may have other locations such as ``"elemental"`` or ``"elementalNodal"``.
+# Note that this way of retrieving data by ID is only valid for locations that
+# are either ``"nodal"`` or ``"elemental"``.
+print(disp_selection.scoping.location)
+
+###############################################################################
+# Another way to check that ``disp_selection`` contains nodal data is to check
+# that the number of entities in the displacement FieldsContainer (12970)
+# matches the number of nodes of the scoped mesh ``mesh_selection``
+print(mesh_selection.nodes.n_nodes)
+
+###############################################################################
+# Instead, the number of elements does not match
+print(mesh_selection.elements.n_elements)
+
+###############################################################################
+# Access data in ``disp_selection`` by ID (index approach)
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Get the list of nodal IDs
+ids = disp_selection.scoping.ids
+
+###############################################################################
+# Retrieve data of ``disp_selection`` from the server. Note that this is done
+# before the loop below so only one call is made to the server.
+start = time.time()
+data = disp_selection.data
+
+###############################################################################
+# Loop over ``ids`` to have access to both, the node ID and the displacement
+# for that node
+nodal_disp_index = np.zeros([mesh_selection.nodes.n_nodes, 3])
+for idx, node_id in enumerate(ids):
+    node_disp = data[idx]
+    nodal_disp_index[idx, :] = node_disp
+
+index_time = time.time() - start
+
+###############################################################################
+# Access data in ``disp_selection`` by ID (get entity approach)
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# An alternative to the index approach is to use the ``get_entity_data_by_id``
+# method to get the nodal displacement given a certain node ID.
+start = time.time()
+nodal_disp_get_entity = np.zeros([mesh_selection.nodes.n_nodes, 3])
+for idx, node_id in enumerate(ids):
+    node_disp = disp_selection.get_entity_data_by_id(node_id)
```

Comment:
@Guillem , this one is useless in this case. the right one in the one you wrote above. 
we use the get_by_id:
-when we loop over several fields for example nodal stresses and displacements and want to get for a given node of the displacement field it's nodal stress for example.
-when we have a user defined scoping, which is different from the field scoping, and we want to extract data from the field. this can be important when we want to read the data once from the file in one request and then work on different named selections and extract data from the field instead of reading each time from a file.

Reply:
Hi @rlagha. Yes, exactly. 

Depending on what you want to do you should use one approach or the other. That is why I included the two of them in the example and a comparison between them.

Reply:
@Guillem, this can be important to mention these remarks in the example.